### PR TITLE
Update Packages in `TestScanSyscollectorCases`

### PR DESCRIPTION
**Pull request by rafabailon**
# Description

Add package metadata to `vuln_packages.json` using results from Research vulnerability Packages used by E2E Vulnerability Detection tests #4943 and edit the cases used by the test.

## Testing Performed

| OS | Wazuh Package | Test Package |
| --- | --- | --- |
| Windows 11 | Agent | NodeJS |
| CentOS 7 | Agent | OpenJDK |
| Ubuntu 22 | Agent | MySQL |

Build: https://ci.wazuh.info/job/Test_e2e_system/243/

| OS  | Result |
| --- | --- |
| Windows 11 | ⚫ |
| CentOS 7 | ⚫ |
| Ubuntu 22 | ⚫ |

### CODE REVIEW

**Comment by rafabailon** [comment]
### Update

I have updated the branch and am trying to run the tests to analyze the results. I cannot use the pipeline due to the limitation in creating MacOS machines. I am trying to set up the necessary machines and launch the tests locally.

**Comment by rafabailon** [comment]
### Update

I have launched the pipeline to build the machines necessary for the tests (except MacOS). I have run the tests locally. When using the testing framework (`wazuh-qa`) I had problems because, although I was using `Python 3.11.6`, when installing the requirements, it installed the wrong version of the `py` package.

https://github.com/wazuh/wazuh-qa/blob/9871e3c1877293bb726cf091ea53facf90cf0e40/requirements.txt#L20
https://github.com/wazuh/wazuh-qa/blob/9871e3c1877293bb726cf091ea53facf90cf0e40/requirements.txt#L21

I have used an Ubuntu mv in version 23.10.

**Comment by rafabailon** [comment]
### Update

You have launched the [pipeline](https://ci.wazuh.info/job/Wazuh_QA_environment/995/) to create the testing environment. I have run the Windows tests.

```console
======================= 7 failed, 10 passed, 11 warnings in 2321.08s (0:38:41) =======================
```

These are the test results: [windows_test.zip](https://github.com/wazuh/wazuh-qa/files/14522366/windows_test.zip)

File: tests/end_to_end/test_vulnerability_detector/cases/test_vulnerability.yaml [hunk 1]
```diff
@@ -104,13 +100,11 @@
           state_index: true
         package:
           centos:
-            amd64: grafana-8.5.5
-            arm64v8: grafana-8.5.5
+            amd64: openjdk-1.6.0"
```
Comment (juliamagan) [inline tests/end_to_end/test_vulnerability_detector/cases/test_vulnerability.yaml:103]:
```suggestion
            amd64: openjdk-1.6.0
```
Those quotation marks are causing an error in the tests, as can be seen in the Linux test report:

```
> package_data = load_packages_metadata()[package_id]
E KeyError: 'openjdk-1.6.0"'
```

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but some changes are required

**Comment by juliamagan** [comment]
In addition, we need to add an entry in the changelog

**Review by juliamagan — APPROVED** [review]
LGTM

**Pull request merged**